src/rapids_singlecell/preprocessing/_sparse_pca/_sparse_pca.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import annotations
+
+import math
+
+import cupy as cp
+
+
+class PCA_sparse:
+    def __init__(self, n_components) -> None:
+        self.n_components = n_components
+
+    def fit(self, x):
```

Comment:
a lot of `fit` seems shared between the dask variant and this, can you share the code?

Reply:
Yes there is. However I want to remove the custom class for the normal sparse PCA early next year. When everyone uses the newer rapids version with the transform from this version.

Reply:
makes sense! if that’s not documented in the file’s docstring, maybe do that.